xdem/spatial_tools.py

```diff
@@ -11,6 +11,55 @@
 import rasterio.warp
 
 
+def get_mask(array: Union[np.ndarray, np.ma.masked_array]) -> np.ndarray:
+    """
+    Return the mask of invalid values, whether array is a ndarray with NaNs or a np.ma.masked_array.
+
+    :param array: Input array.
+
+    :returns invalid_mask: boolean array, True where array is masked or Nan.
+    """
+    return (array.mask | np.isnan(array.data)) if isinstance(array, np.ma.masked_array) else np.isnan(array)
+
+
+def get_array_and_mask(array: Union[np.ndarray, np.ma.masked_array]) -> (np.ndarray, np.ndarray):
```

Comment:
Same here with the underscore to hide it from the public API

Reply:
Decided to leave it as it is for now.